pd/src/state.rs

```diff
@@ -39,6 +42,83 @@ impl State {
         Ok(State { pool })
     }
 
+    /// Commits the genesis config to the database, prior to the first block commit.
+    pub async fn commit_genesis(&self, genesis_config: &genesis::AppState) -> Result<()> {
+        let mut dbtx = self.pool.begin().await?;
+
+        // ON CONFLICT is excluded here so that an error is raised
+        // if genesis config is attempted to be set more than once
+        query!(
+            r#"
+INSERT INTO blobs (id, data) VALUES ('gc', $1)
+"#,
+            &serde_json::to_vec(&genesis_config)?[..]
+        )
+        .execute(&mut dbtx)
+        .await?;
+
+        // TODO: move into BaseRateData::at_genesis() ctor?
+        query!(
+            "INSERT INTO base_rates (epoch, base_reward_rate, base_exchange_rate) VALUES (0, 0, $1)",
+            1_0000_0000
+        ).execute(&mut dbtx).await?;
+
+        for genesis::ValidatorPower { validator, power } in &genesis_config.validators {
+            query!(
+                "INSERT INTO validators (
+                    identity_key, 
+                    consensus_key, 
+                    sequence_number, 
+                    validator_data
+                ) VALUES ($1, $2, $3, $4)",
+                validator.identity_key.to_bytes().to_vec(),
+                validator.consensus_key.to_bytes(),
+                validator.sequence_number as i64,
```

Comment:
Here and elsewhere, I changed the state code to do the `i64`/`u64` conversion as close to the DB layer as possible.  It's annoying that Postgres doesn't support unsigned integers, but it seems better to avoid propagating that annoyance through all of our data structures.

This conversion is lossless (it's just changing the interpretation of the sign bit), so we'll always get the data back; the only risk is if we try to do arithmetic inside of Postgres on values in excess of 2^63. The only place that it seems likely we'd do arithmetic inside of Postgres is something like sorting, but values we'd be likely to sort on (like block heights) probably won't be brushing against the top of the value range anyways.